### PR TITLE
Correct documentation when calling for_controller

### LINKED ISSUE #5: Passing view context to presenter in tests
Hi @andypike !

Quick question. How can I test presenters that rely on view context? My test is failing because the path helpers are undefined for the presenter in the test. They work fine in the browser.

```
Failure/Error: "#{ hospital.name } - #{ link_to 'View', hospital_path(hospital) }"

     NoMethodError:
       undefined method `hospital_path' for #<DeaneryPresenter:0x007fe0afcca1b8>
```

I've tried adding `include ActionView::TestCase::Behavior` and using `.for_controller(controller)` in the spec file but then it complains that the methods defined in my presenter are undefined for the controller
```
undefined method `map_markers' for #<ActionView::TestCase::TestController:0x007fbc48b135d0>
```

Thanks in advance for any help! :)

It will work if I explicitly include everything in the presenter:
```
class DeaneryPresenter < Rectify::Presenter
  include Rails.application.routes.url_helpers
  include ActionView::Helpers::UrlHelper
  include ActionView::Helpers::TagHelper
```

### PR DESCRIPTION
**Pull request by kimberger**
Closes #5 . 
```
@presenter = UserDetailsPresenter.new(:user => user).for_controller(self)
```
Will set `@presenter` to be equal to the controller, rather than the presenter. The change suggested is consistent with the [Presenter Specs](https://github.com/andypike/rectify/blob/master/spec/lib/rectify/presenter_spec.rb#L27-L28)

### CODE REVIEW

**Comment by andyw8** [comment]
I think the naming of `for_controller` is a little confusing, perhaps it should be something like `attach_controller`?

Alternatively, maybe `for_controller` could be a factory method which returns a new instance, e.g.:

```
presenter = UserDetailsPresenter.for_controller(self, :user => user)
```

**Comment by andypike** [comment]
Thanks @kimberger and @andyw8! 

I think that `for_controller` should return the presenter so the original documented example actually works (sorry about that). I'm all for improving naming and am happy to go with `attach_controller` as I agree that's better (unless you have an even better option?).

I'm not sure we need to make it a factory method as the normal usage should be to include `Rectify::ControllerHelpers` in your controller and then use the `present` method which will call `attach_controller` for you.

To summarise:

1. Rename `Rectify:: Presenter#for_controller` to `Rectify:: Presenter#attach_controller` and update all references
2. `Rectify:: Presenter#attach_controller` should return the presenter (`self`)

How you feel about those? @kimberger would you like to give those changes a try? If not, I'm happy to do them, just let me know :smile:

Thanks again for the feedback :heart:

**Comment by andyw8** [comment]
@andypike: I see. So `#for_controller` would likely only ever be used in a test, but not in production code? If so, then maybe it shouldn't be possible to instantiate a **without** a controller, i.e. it should be a required params of the initializer? This wouldn't affect the existing syntax of the `present` method but should help steer the tester in the right direction.

**Comment by kimberger** [comment]
@andypike  Happy to do those changes. I did consider having `for_controller` return the presenter instead but given the code in the test and that the doc change was less intrusive I went for the doc change instead. :smile:

**Comment by andypike** [comment]
@andyw8 That's right, I would expect that in production most people won't use `#for_controller` directly but I wanted to document it in case anyone prefers to avoid using the helper methods (and of course in tests).

I think I would prefer not to force the controller to be required at this stage because there may be some presenters that don't need access to the `view_context`. Only presenters that need view helper methods will need to specify a controller. I suspect those presenters will be rare (as a decorator is probably a better fit) but I think it's valid for a presenter exist without access to a `view_context`. 

Does that sound ok or do you think we should enforce the usage patterns more. By that I mean, presenters are for view logic with use of view helper methods and decorators are for view logic without view helpers? Interested in your thoughts. :heart:

**Comment by kimberger** [comment]
Those changes are going in. I also changed the spec so that it's consistent with the docs but breaks 80 char line limit. Happy to change it back if you'd like.

My preference would be to leave the view_context as optional. Doesn't seem right to force the juggling of Decorators and Presenters for the sake of including or excluding view_context?

**Comment by andyw8** [comment]
@andypike Ok, makes sense. :+1:

**Comment by andypike** [comment]
Thanks chaps! @kimberger would you mind adding a spec to ensure that `#attach_controller` returns the presenter? I think we are good to merge once that's in.

Thanks again for the feedback, discussion and the PR! :heart: :sparkles: :+1:

**Comment by andypike** [comment]
@kimberger Oh, and how do you feel about squashing these commits down into one?

Cheers

**Comment by kimberger** [comment]
Don't be sorry. You're right `eq(presenter)` is much more specific. Thanks for the advice :+1:  

Looking forward to integrating this into my own app.

**Comment by andypike** [comment]
Thanks for the PR!! 

:sparkles: :star: :shipit: :star: :sparkles:
